apis/route53resolver/v1alpha1/zz_resolver_rule.go

```diff
@@ -0,0 +1,146 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Code generated by ack-generate. DO NOT EDIT.
+
+package v1alpha1
+
+import (
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// ResolverRuleParameters defines the desired state of ResolverRule
+type ResolverRuleParameters struct {
+	// Region is which region the ResolverRule will be created.
+	// +kubebuilder:validation:Required
+	Region string `json:"region"`
+	// A unique string that identifies the request and that allows failed requests
+	// to be retried without the risk of executing the operation twice. CreatorRequestId
+	// can be any unique string, for example, a date/time stamp.
+	// +kubebuilder:validation:Required
+	CreatorRequestID *string `json:"creatorRequestID"`
+	// DNS queries for this domain name are forwarded to the IP addresses that you
+	// specify in TargetIps. If a query matches multiple Resolver rules (example.com
+	// and www.example.com), outbound DNS queries are routed using the Resolver
+	// rule that contains the most specific domain name (www.example.com).
+	// +kubebuilder:validation:Required
+	DomainName *string `json:"domainName"`
+	// A friendly name that lets you easily find a rule in the Resolver dashboard
+	// in the Route 53 console.
+	Name *string `json:"name,omitempty"`
```

Comment:
Is this different than `ResolverRuleId`? In some cases, AWS uses id and name interchangeably, so if they correspond to the same thing we should ignore this field and use external name annotation as its value, hence remove `SetExternalName` from `postCreate` and use `NewNameAsExternalName` initializer.

Reply:
@muvaf  The id and name for both resolver endpoint and resolver rule correspond to different things.
https://docs.aws.amazon.com/sdk-for-go/api/service/route53resolver/#ResolverRule
https://docs.aws.amazon.com/sdk-for-go/api/service/route53resolver/#ResolverEndpoint